### ISSUE TITLE
How to interpret HMM states without labeling?

### ISSUE THREAD

**Issue by adis86** [hmmlearn/hmmlearn#229]
First of all, thanks for sharing the work on HMM, it's really usefull. 

My question is as follows. I have a data set which consists of positive floating-point numbers: `X = [x1, x2, ..., xn]`. Then I use this array to train a GMMHMM model. 

`gmmhmm_model = hmmlearn.hmm.GMMHMM(n_components=2, n_mix=3, covariance_type="full", algorithm="viterbi", init_params="stcmw", params="stcmw", n_iter=20, tol=1e-4)`

```
X = numpy.array(X).reshape(-1, 1)
gmmhmm_model.fit(X, length=[X.shape[0]])
```

Then, I use predict function to get a sequence of states given test data Y which is a set of positive floating-point numbers like X. 

```
Y = numpy.array(Y).reshape(-1, 1)
seq = gmmhmm_model.predict(Y, lengths=[Y.shape[0]])
```

I will get a sequence of 0s and 1s which correspond to the two HMM states I defined. But my question is how do you interpret these states as you don't use labels when training. Can I say, state_0 always represents small numbers and vice versa?

**Comment by jexner** [comment]
I think this is a case of [label swtiching problem](https://stackoverflow.com/questions/39756006/how-to-map-hidden-states-to-their-corresponding-categories-after-decoding-in-hmm). You could create a labelled sequence, feed it to your trained HMM and use the predictions to infer the mapping from state index to label.
